### ISSUE TITLE
Add resource limits configuration capabilities for the chaincode pods

### ISSUE THREAD

**Issue by davidfdr** [hyperledger-labs/fabric-builder-k8s#102]
The chaincode PODS don ´t have defined the resource limits.

Those limits can be configured in:

1. Chaincode Artifacts - image.json;
2. Environment Variables;
4. etc?

**Comment by jt-nti** [comment]
Defining resource limits definitely seems like a good idea. Configuration for the builder is done via environment variables since the chaincode package should be what is being deployed, not how it is deployed. Extra environment variables for everything you might want to configure in a pod doesn't seem like a good idea though. I'm wondering if using a ConfigMap might work, or if there is some other/better way to template pod configuration in kubernetes.

**Comment by davidabinbev** [comment]
@jt-nti  One question:

1. Did we will specify resource limits per chaincode or all chaincodes will have the same resource limits?

If all chaincodes use the same reource limits definition:

I prefer using environment variables then using config maps. Less k8s configuration templates I think that is better for a easy adoption (Less configuration points / centralized configuration point).

If we will set the resource configurations per chaincode, maybe using the image.json file is better.

Thoughts?

**Comment by davidfdr** [comment]
> Defining resource limits definitely seems like a good idea. Configuration for the builder is done via environment variables since the chaincode package should be what is being deployed, not how it is deployed. Extra environment variables for everything you might want to configure in a pod doesn't seem like a good idea though. I'm wondering if using a ConfigMap might work, or if there is some other/better way to template pod configuration in kubernetes.

@jt-nti one question:

What is better, resource limist set by chaincode or all chaincodes runnning with the same resource limits?

If all chaincodes will be using the same resouce limits, I prefer setting this in a ENVIRONMENT VARIABLE. I know that we can use config maps, but, I prefer less K8S config points and we already have the environment variables. I think that a centralized configuration point is better for the builder adoption.

If we want to have the resouce limits configured per chaincode, I think that the best location will be the image.json artifact file.

Thoughts?

**Comment by jt-nti** [comment]
Agreed, I think it does make sense to use environment variables. There are a few different options though:
- a high level small/medium/large style setting to select between some preset resource requests and limits
- separate environment variables for cpu/memory requests/limits
- a single environment variable for all the requests and limits using a separator or possibly even json

I've opened #104 to explore how to handle different settings for different chaincode in the future.

**Comment by davidfdr** [comment]
I think:

- separate environment variables for cpu/memory requests/limits

I think that It's easiear to handle and a more direct approach. 

A high level small/medium/large approach will have the need of a good documentation and explanation.

Anyway, I agree with the first two approaches.

**Comment by labibfarag** [comment]
how to control the  Env variables value per the chaincode 
chaincode A requires high CPU and memory 
but chaincode B requires low memory and low CPU
if env variables are used will be the same case as  all chaincodes running on the same namespace

**Comment by jt-nti** [comment]
@davidfdr separate environment variables for cpu/memory requests/limits sound good to me

> how to control the Env variables value per the chaincode
chaincode A requires high CPU and memory
but chaincode B requires low memory and low CPU
if env variables are used will be the same case as all chaincodes running on the same namespace

@labibfarag I've opened #104 to investigate this kind of requirement. At the moment you would need to install chaincode A and B on different peers to be able to configure a different namespace.

**Comment by jt-nti** [comment]
@davidfdr are you still planning to work on this issue? If not, I'll unassign it.

**Comment by jt-nti** [comment]
Superseded by #630
